framework/wazuh/engine/request_message.py

```diff
@@ -0,0 +1,101 @@
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+from typing import Dict, Any, Optional
+from wazuh.engine.commands import EngineCommand
+
+
+class EngineRequestMessage:
+    """ Represents a request to the engine"""
+    def __init__(self, version: int):
+        """
+        Initialize the EngineRequestBuilder object.
+
+        Parameters
+        ----------
+        version : str
+            The version of the engine request.
+        """
+        self.internal_dict: Dict[str, Any] = {"version": version}
+
+    def create_message(self, origin_name: str, module: str, command: EngineCommand,
+                       parameters: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
+        """
+        Create a message for the engine request.
+
+        Parameters
+        ----------
+        origin_name : str
+            The name of the origin.
+        module : str
+            The module of the origin.
+        command : EngineCommand
+            The command for the engine request.
+        parameters : Dict[str, Any], optional
+            Dictionary with the parameters to add, by default None
+
+        Returns
+        -------
+        Dict[str, Any]
+            The created engine request message.
+
+        """
+
+        self.add_origin(name=origin_name, module=module)
+        self.add_command(command=command)
+
+        self.add_parameters(parameters=parameters if parameters else {})
+
+        return self.to_dict()
+
+    def add_origin(self, name: str, module: str):
+        """
+        Add the origin details to the engine request.
+
+        Parameters
+        ----------
+        name : str
+            The name of the origin.
+        module : str
+            The module of the origin.
+        """
+        self.internal_dict["origin"] = {
+            "name": name,
+            "module": module
+        }
+
+    def add_command(self, command: EngineCommand):
+        """
+        Add the command to the engine request.
+
+        Parameters
+        ----------
+        command : EngineCommand
+            The command for the engine request.
+        """
+        self.internal_dict["command"] = command.value
+
+    def add_parameters(self, parameters: Dict[str, Any]):
+        """
+        Add a parameter to the engine request.
+
+        Parameters
+        ----------
+        parameters: Dict[str, Any]
+            Dictionary with the parameters to add
+        """
+        self.internal_dict.setdefault("parameters", {})
+        self.internal_dict["parameters"].update(parameters)
+
+    def to_dict(self) -> Dict[str, Any]:
+        """
+        Convert the EngineRequestBuilder object to a dictionary representation.
+
+        Returns
+        -------
+        Dict[str, Any]
+            The dictionary representation of the EngineRequestBuilder object.
+        """
+        return self.internal_dict
```

Comment:
I think it would be nice to keep the `EngineRequestMessage` class for the following reasons:

- Although the message is sent over a socket, the functionality in `wazuh_socket` is generic while that of `EngineRequestMessage` is Engine specific and abstracts from the developer what format the message should have.
- The version in `wazuh_socket` is hardcoded and represents the version used when communicating with sockets. In case the Engine changes the version, we would have to find a way to change this hardcoded version without changing all the other endpoints/controllers that use the hardcoded version. Instead, by having a specialized class for the Engine, we would not have to see how to implement this change.
- I understand the logic of using a functionality that is already tested, but since it is a fairly simple logic (a wrapper for a dictionary with a specific format), it doesn't seem to me that the `EngineRequestMessage` class introduces a very difficult logic to test or introduces some bug without realizing it

This is just my opinion, it is a library that all of us who develop with the new Engine are going to use. If despite these points it still seems more convenient not to use it, I delete it without a problem.

Reply:
I understand your points but I would still prefer to use the existing implementation to keep things simple and to avoid repeated code. If there is anything `wazuh_socket` is missing I would try to add the functionality to it (i.e. the engine commands could have a version that is different from the one the other commands use).

We could share our thoughts with the team to decide how to proceed.